### PR TITLE
Python 3: Remove autotest dependency

### PR DESCRIPTION
**Pull request by czhang0**
Signed-off-by: cuzhang <cuzhang@redhat.com>

### CODE REVIEW

**Comment by czhang0** [comment]
Take virsh_node_memtune as example to test, test result is below:

#avocado run --vt-type libvirt type_specific.local.virsh.node_memtune
JOB ID     : 6702fdc357148aab50b80fd0a795399717e36f68
JOB LOG    : /root/avocado/job-results/job-2018-03-05T11.22-6702fdc/job.log
 (01/15) type_specific.local.virsh.node_memtune.positive_testing.get_node_memory_parameter: PASS (1.33 s)
 (02/15) type_specific.local.virsh.node_memtune.positive_testing.set_node_memory_parameter.shm_pages_to_scan.normal: PASS (1.31 s)
 (03/15) type_specific.local.virsh.node_memtune.positive_testing.set_node_memory_parameter.shm_pages_to_scan.zero: PASS (1.33 s)
 (04/15) type_specific.local.virsh.node_memtune.positive_testing.set_node_memory_parameter.shm_sleep_millisecs.normal: PASS (1.31 s)
 (05/15) type_specific.local.virsh.node_memtune.positive_testing.set_node_memory_parameter.shm_sleep_millisecs.zero: PASS (1.33 s)
 (06/15) type_specific.local.virsh.node_memtune.positive_testing.set_node_memory_parameter.shm_merge_across_nodes.disable: PASS (1.31 s)
 (07/15) type_specific.local.virsh.node_memtune.positive_testing.set_node_memory_parameter.shm_merge_across_nodes.enable: PASS (1.33 s)
 (08/15) type_specific.local.virsh.node_memtune.negative_testing.get_node_memory_parameter: PASS (1.31 s)
 (09/15) type_specific.local.virsh.node_memtune.negative_testing.set_node_memory_parameter.invalid_value.shm_pages_to_scan: PASS (1.33 s)
 (10/15) type_specific.local.virsh.node_memtune.negative_testing.set_node_memory_parameter.invalid_value.shm_sleep_millisecs: PASS (1.30 s)
 (11/15) type_specific.local.virsh.node_memtune.negative_testing.set_node_memory_parameter.invalid_value.shm_merge_across_nodes.invalid_number: PASS (1.33 s)
 (12/15) type_specific.local.virsh.node_memtune.negative_testing.set_node_memory_parameter.invalid_value.shm_merge_across_nodes.invalid_operation: PASS (1.32 s)
 (13/15) type_specific.local.virsh.node_memtune.negative_testing.set_node_memory_parameter.invalid_options.option.shm_pages_to_scan: PASS (1.32 s)
 (14/15) type_specific.local.virsh.node_memtune.negative_testing.set_node_memory_parameter.invalid_options.option.shm_sleep_millisecs: PASS (1.33 s)
 (15/15) type_specific.local.virsh.node_memtune.negative_testing.set_node_memory_parameter.invalid_options.option.shm_merge_across_nodes: PASS (1.31 s)
RESULTS    : PASS 15 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 24.70 s

File: libvirt/tests/src/virsh_cmd/domain/virsh_save.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import os
 
-from autotest.client.shared import error
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_save.py]: please delete one more blank line

File: libvirt/tests/src/virsh_cmd/domain/virsh_schedinfo_qemu_posix.py [hunk 2]
```diff
@@ -2,7 +2,6 @@
 import logging
 import os
 
-from autotest.client.shared import error
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_schedinfo_qemu_posix.py]: please delete one more blank line

File: libvirt/tests/src/virsh_cmd/domain/virsh_setmem.py [hunk 3]
```diff
@@ -3,7 +3,6 @@
 import logging
 import time
 
-from autotest.client.shared import error
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setmem.py]: delete one more blank line

File: libvirt/tests/src/virsh_cmd/host/virsh_node_memtune.py [hunk 4]
```diff
@@ -95,22 +94,27 @@ def get_node_memtune_parameter(params):
 
     # Check status_error
     status_error = params.get("status_error", "no")
+    test_fail = False
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/host/virsh_node_memtune.py]: I don't think you need change previous implementation logic

File: libvirt/tests/src/virsh_cmd/host/virsh_node_memtune.py [hunk 5]
```diff
@@ -183,39 +192,31 @@ def run(test, params, env):
 
     if status_error == "no":
         if change_parameters == "no":
-            try:
-                get_node_memtune_parameter(params)
-            except error.TestFail, detail:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/host/virsh_node_memtune.py]: changed to except exceptions.TestFail, detail:

File: libvirt/tests/src/virsh_cmd/network/virsh_net_dumpxml.py [hunk 6]
```diff
@@ -67,7 +66,7 @@ def run(test, params, env):
         status = result.exit_status
         err = result.stderr.strip()
         xml_validate_cmd = "virt-xml-validate %s network" % network_xml
-        valid_s = utils.run(xml_validate_cmd, ignore_status=True).exit_status
+        valid_s = process.run(xml_validate_cmd, ignore_status=True).exit_status
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_dumpxml.py]: need set shell=True

File: libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml.py [hunk 7]
```diff
@@ -49,7 +49,7 @@ def dump_nodedev_xml(dev_name, dev_opt="", **dargs):
         # nodedev_dict_sys contain the all keys and values in sysfs.
         nodedev_dict_sys = {}
         for key, filepath in nodedev_syspath_dict.items():
-            value = utils.read_one_line(filepath)
+            value = process.run(filepath, shell=True).stdout.split()[0]
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml.py]: utils.read_one_line in autotest implement as:
open(filename, 'r').readline().rstrip('\n')

Not sure process.run can deal with it gracefully, so better replace it with open(filename, 'r').readline().rstrip('\n')

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py [hunk 8]
```diff
@@ -343,11 +342,8 @@ def check_pool_info(pool_info, check_point, value):
         check_pool_list(pool_name, "--all", True)
     finally:
         # Clean up
-        try:
-            pvt.cleanup_pool(pool_name, pool_type, pool_target,
-                             emulated_image, **kwargs)
-        except error.TestFail, detail:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool.py]: change to  except exceptions.TestFail, detail:

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py [hunk 9]
```diff
@@ -343,11 +342,8 @@ def check_pool_info(pool_info, check_point, value):
...
-            logging.error(str(detail))
+        pvt.cleanup_pool(pool_name, pool_type, pool_target,
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool.py]: Better keep previous implementation

File: libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot.py [hunk 10]
```diff
@@ -140,8 +138,6 @@ def check_info(i1, i2, errorstr="Values differ"):
 
         except process.CmdError:
             handle_error("Failed getting snapshots info", vm_name)
-        except error.TestFail, e:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot.py]: except exceptions.TestFail, e:

File: libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot.py [hunk 11]
```diff
@@ -167,11 +163,9 @@ def check_info(i1, i2, errorstr="Values differ"):
             test_file(session, sni["to_delete"], 2)
         except process.CmdError:
             handle_error("Failed to revert snapshot", vm_name)
-        except (error.TestFail, virt_vm.VMDeadError), e:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot.py]: change  error.TestFail to exceptions.TestFail

File: libvirt/tests/src/virsh_cmd/domain/virsh_save.py (hunk 1 above)
Comment (czhang0) [reply libvirt/tests/src/virsh_cmd/domain/virsh_save.py]: Adopt.

**Pull request merged**